src/CDN/CloudFrontVersion3.php

```diff
@@ -0,0 +1,163 @@
+ * If you need to invalidate a lot of files, it might be easier and less
+ * expensive to create a new distribution and change your object paths to refer
+ * to the new distribution. For more information about the charges for
+ * invalidation, see Paying for Object Invalidation.
+ *
+ * @uses CloudFrontClient for establish connection with CloudFront service
+ *
+ * @see https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/Invalidation.htmlInvalidating Objects (Web Distributions Only)
+ *
+ * @author Javier Spagnoletti <phansys@gmail.com>
+ */
+final class CloudFrontVersion3 implements CDNInterface
+{
+    private const AVAILABLE_STATUSES = [
+        self::STATUS_OK => 'Completed',
+        self::STATUS_TO_SEND => 'STATUS_TO_SEND',
+        self::STATUS_TO_FLUSH => 'STATUS_TO_FLUSH',
+        self::STATUS_ERROR => 'STATUS_ERROR',
+        self::STATUS_WAITING => 'InProgress',
+    ];
+
+    /**
+     * @var CloudFrontClient
+     */
+    private $client;
+
+    /**
+     * @var string
+     */
+    private $distributionId;
+
+    /**
+     * @var string
+     */
+    private $path;
+
+    public function __construct(CloudFrontClient $client, string $distributionId, string $path)
+    {
+        $this->client = $client;
+        $this->distributionId = $distributionId;
+        $this->path = rtrim($path, '/');
+    }
+
+    public function getPath($relativePath, $isFlushable = false): string
+    {
+        return sprintf('%s/%s', $this->path, ltrim($relativePath, '/'));
+    }
+
+    public function flushByString($string): string
+    {
+        return $this->flushPaths([$string]);
+    }
+
+    public function flush($string): string
+    {
+        return $this->flushPaths([$string]);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @see https://docs.aws.amazon.com/aws-sdk-php/latest/class-Aws.CloudFront.CloudFrontClient.html#_createInvalidation
+     */
+    public function flushPaths(array $paths): string
+    {
+        if (empty($paths)) {
+            throw new \RuntimeException('Unable to flush : expected at least one path');
+        }
+        // Normalizes paths due possible typos since all the CloudFront's
+        // objects starts with a leading slash
+        $normalizedPaths = array_map(static function (string $path): string {
+            return '/'.ltrim($path, '/');
+        }, $paths);
+
+        try {
+            $result = $this->client->createInvalidation([
+                'DistributionId' => $this->distributionId,
+                'InvalidationBatch' => [
+                    'Paths' => [
+                        'Quantity' => \count($normalizedPaths),
+                        'Items' => $normalizedPaths,
+                    ],
+                    'CallerReference' => $this->getCallerReference($normalizedPaths),
+                ],
+            ]);
+            $status = $result->get('Invalidation')['Status'];
+
+            if (!\in_array($status, ['Completed', 'InProgress'], true)) {
```

Comment:
Why don't you use your defined constants? 
```suggestion
            if (!\in_array($status, [self::AVAILABLE_STATUSES[self::STATUS_OK], self::AVAILABLE_STATUSES[self::STATUS_WAITING]], true)) {
```